instrumentation_sql.go

```diff
@@ -227,11 +227,38 @@ func (conn *wrappedSQLConn) ExecContext(ctx context.Context, query string, args
 }
 
 func (conn *wrappedSQLConn) CheckNamedValue(d *driver.NamedValue) error {
+	defer func() {
+		if err := recover(); err != nil {
+			sensor.logger.Debug("Swallowing error: ", err)
+		}
+	}()
+
 	if c, ok := conn.Conn.(driver.NamedValueChecker); ok {
 		return c.CheckNamedValue(d)
 	}
 
-	return nil
+	q := "create table instanatemp (a varchar(1))"
+
+	stmt, err := conn.Prepare(q)
+
+	if err != nil {
+		sensor.logger.Debug("Database does not support temporary statement: ", q)
+	}
+
+	defer stmt.Close()
+
+	if s, ok := stmt.(driver.NamedValueChecker); ok {
+		return s.CheckNamedValue(d)
+	}
+
+	d.Value, err = driver.DefaultParameterConverter.ConvertValue(d.Value)
+
+	if err != nil {
```

Comment:
In the case that the Go default method was used and it failed, we log the error but return nil, to prevent customers code to break.
Most likely it should not come here, but if it happens, chances are that we need to cover a new case based on a specific driver. eg: a driver that does conversions of more complex types, other than the ones checked by the standard lib.